apps/openmw/mwworld/weather.cpp

```diff
@@ -663,6 +665,40 @@ void WeatherManager::playerTeleported(const std::string& playerRegion, bool isEx
     }
 }
 
+osg::Vec3f WeatherManager::calculateStormDirection()
+{
+    osg::Vec3f playerPos (MWMechanics::getPlayer().getRefData().getPosition().asVec3());
+    playerPos.z() = 0;
+    osg::Vec3f redMountainPos (25000, 70000, 0);
+    osg::Vec3f stormDirection = (playerPos - redMountainPos);
+    stormDirection.normalize();
+
+    return stormDirection;
+}
+
+float WeatherManager::calculateWindSpeed(int weatherId, float currentSpeed)
+{
+    float targetSpeed = std::min(8.0f * mWeatherSettings[weatherId].mWindSpeed, 70.f);
+    if (currentSpeed == 0.f)
+        currentSpeed = targetSpeed;
```

Comment:
This is probably okay for gameplay, but the stuff done in this function isn't exactly linear interpolation. Should this particular line from "zero to targetspeed" be smoother? How about 70.f? How come that is the minimum?

Reply:
This is just the values from [reverse-engineered vanilla research](https://wiki.openmw.org/index.php?title=Research:Weather#Wind_speed). Any constant names will have to be pulled out of the backside.

What do you mean "smoother"? There's no gameplay-side transition from 0 to the precalculated speed.